proof/refine/ARM/Tcb_R.thy

```diff
@@ -24,47 +24,262 @@ lemma activate_idle_thread_corres:
     (arch_activate_idle_thread t) (activateIdleThread t)"
   by (simp add: arch_activate_idle_thread_def activateIdleThread_def)
 
+lemma gts_st_tcb':
+  "\<lbrace>tcb_at' t\<rbrace> getThreadState t \<lbrace>\<lambda>rv. st_tcb_at' (\<lambda>st. st = rv) t\<rbrace>"
+  apply (rule hoare_vcg_precond_imp)
+   apply (rule hoare_post_imp[where Q="\<lambda>rv s. \<exists>rv'. rv = rv' \<and> st_tcb_at' (\<lambda>st. st = rv') t s"])
+    apply simp
+   apply (wp hoare_ex_wp)
+  apply (clarsimp simp add: pred_tcb_at'_def obj_at'_def)
+  done
+
+lemma activateIdle_invs:
```

Comment:
`invs'` instead? (Though the consistency with this is currently quite poor in Refine)